lblod/worship-submissions-email-notification-service
# Improving email notification service

# Description
DL-5742 -> DL-5715

This PR fixes some issues with the service, updating dependencies along with adding more information about submissions in the email template. 

- Updating dependencies
- Adding date-fns and date-fns-tz packages
- Adjusting query variable `?submission` in favor of `?submissionUri`
- Removing mock-email feature		
- Adding formatted url directly into the submission object from each submissions with the function  `addUrlPerSubmission(submissionUri)` previously named `getLinks()` https://github.com/lblod/worship-submissions-email-notification-service/pull/3/files#diff-80b0520dd0dbcf7162aba6a748e053a5ab7d2f96603a96e60fc710249f535dbfL133
- Improving conditional check, submissions wont be appended twice and if submissions made by `creatorEenheidLabel` = `targetEenheidLabel`, they wont be added to the bundle
- Adjusting HTML template
  - Adding style to include a table
  - Injecting table rows with each submissions information such as `sentDate`, `decisionTypeLabel`, `creatorEenheidLabel` and `url`
- Adding plain text email in DB

## Template preview v1.2.0-rc.1
![Screenshot 2024-04-02 at 16-07-17 Screenshot](https://github.com/lblod/worship-submissions-email-notification-service/assets/38471754/d9040352-f282-4ab8-9376-75474f7bd0f4)

## Template preview with https://github.com/lblod/worship-submissions-email-notification-service/pull/3/commits/7108ad56d78b1190f42d374ad5e5d5e85d6bc24b https://github.com/lblod/worship-submissions-email-notification-service/pull/3/commits/21262afbb335678393f3f7e58805beb5fa795453

![Screenshot 2024-04-04 at 14-05-30 Screenshot](https://github.com/lblod/worship-submissions-email-notification-service/assets/38471754/5b239186-a1ee-4b34-8d49-63f85bff7084)


# Type of change


# Related stack

- https://github.com/lblod/app-worship-decisions-database

# How to test 

1. Run the worship-decisions-database stack with the new image. Add a config if you didn't, see https://github.com/lblod/worship-submissions-email-notification-service?tab=readme-ov-file#-quick-setup
2. Subscribe to notifications 
![Screenshot 2023-09-18 at 17-17-13 Erediensten Toezichtsdatabank-inzending](https://github.com/lblod/worship-submissions-email-notification-service/assets/38471754/13bd5cc5-4e75-4d1f-befb-2491a4ac9771)
3. Log the service, it should bundle notifications to be sent. Few rules are executed.
3.1. Submissions sent by the target (subscribed administrative unit) wont be sent as notification
3.2. Submissions shouldn't be duplicated in the notifications, you should end up with this message after the process is done `Placing ${targetEenheidLabel}'s email notification ${email.uri} into outbox` as reminder outbox = `<http://data.lblod.info/id/mail-folders/2>`
4. Log `deliver-email-service | grep 'Preview' ` it should return an URL to preview the email locally meaning that email is processed and delivered in sentbox.

# What to check

- Any typo or styling issues with the email template

# Links to other PR's

- N/A

# Notes

Release and Bump docker image in app-worship-decisions-database when merged

drc up -d worship-submission-email-notification-service
